### PR TITLE
Function to download history data

### PR DESCRIPTION
**Pull request by llrs**
This is a draft after some iterations on the function (I originally used bash).
It depends on the number of columns on the file to simplify and merge the different files. 

I hope it is clear enough (but maybe it could be simplified with a helper function).
It also iterates two times through the list of .csv files, that could be reduced but increasing the memory required earlier. (I don't think this is an issue and it could help if column names are reordered).
It doesn't check if data was recently downloaded or cache the data.

Let me know how do you see it and I'll finish writing the documentation.

### CODE REVIEW

**Comment by Bisaloo** [comment]
Before I go into the actual review, I have a couple of questions for you:

- Do you think we should reformat older files to make sure all files use the same format?
- Should we remove the non-csv files from the `history` branch? I believe `cran-diagram.png` was added by mistake and Hadley's `analysis.R` could be promoted to the `main` branch alongside your function to make it more discoverable.

**Comment by llrs** [comment]
Good questions.
- I don't like to tamper with original files. I would leave them as they are. Later on in a couple of years maybe the function could be modified to load data from a certain date (or last X period) which would exclude these files. 
- That might be useful but they aren't much of a problem.  I don't think the package is the right place to share analysis code on  an article or branch. But sharing a minimal insight from the data collected might be useful for users of the dashboard (I would report numbers but hide or fold the code used).

**Review by Bisaloo — COMMENTED** [review]
Looks good to me. 

A couple of points to update:
-  [ ] I'd like more comments about the various formats and when they were in use. I'll submit commit suggestions for this later
- [ ] Could you rename the file to `download_history.R` please?
- [x] Could you add a roxygen title?
- [x] Could you add a roxygen comment with `#' @importFrom utils download.file read.csv unzip`?
- [x] Once the two points above have been addressed, could you run `devtools::document()`?

I'm happy to post commit suggestions or do the changes myself if convenient. Please let me know :relaxed:

**Comment by llrs** [comment]
Sorry, I almost forgot about this.
I'll do the changes and update the documentation (probably this week). But if you want to provide suggestion on the comments/dates go ahead.

**Comment by llrs** [comment]
I think I addressed all the point raised. I took the liberty to use `desc::desc_normalize` to arrange the order of dependencies after adding utils.

**Comment by llrs** [comment]
Gentle(?) reminder @Bisaloo and @maelle  (I just remember about this and wasn't sure if I had updated the PR or not)

**Review by Bisaloo — COMMENTED** [review]
Oh, I missed the fact you added comments with the dates so I wrote my own. I tried to merge what we both did. Please let me know what you think.

One final comment: I just realized now you're not in the authors of this package. Could you add yourself as `aut` please?

File: R/download_history.R [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  tmp_dir <- tempdir()
+  download.file("https://github.com/r-hub/cransays/archive/history.zip",
+                destfile = tmp_f)
+  # We unzip the files
+  dat <- unzip(tmp_f, exdir = tmp_dir, setTimes = TRUE)
+  dat <- dat[endsWith(dat, ".csv")]
+
+  # First two heading systems:
+  incoming_1 <- dat[startsWith(basename(dat), "cran-incoming_-")]
+  # Header used 2020-09-12 till 2020-09-14
```
[the remark this answers is not in the document]
Comment (llrs) [reply R/download_history.R]: Moved the line. Indeed it is easier if the headers were chronological. I didn't knew which one was first before your request and then I didn't want to introduce a new change after your first review. But I have reordered this too.

**Comment by llrs** [comment]
:+1:  I fixed the comments, reordered the code and I added myself as an author

**Review by Bisaloo — APPROVED** [review]

**Comment by Bisaloo** [comment]
Thanks!

**Pull request merged**